Travis CI Integration

jplyle opened the pull request:
http://jira.webinos.org/browse/WP-793

This pull request contains all the configuration needed to get the Webinos Project using Travis CI for testing.  It also contains a few minor edits to tests and to the PZP to enable better testing.

Commit:
Initial travis-ci testing files

This commit also comments out a syntax error in the
SensorSpec test and adds jasmine-node to the list
of dependencies.

Jira issue: WP-793

Commit:
Travis CI testing: fix .yml and package.json

Due to grunt changes and misconfiguration in the
earlier commit, the travis-ci build wasn't working.  This
commit changes .travis.yml and adds the right grunt
dependencies.

Commit:
More travis.yml configuration for continuous build.

This time: made npm be more verbose and test both
node 0.6 and 0.8.  Also added node-gyp dependency.

Jira issue: WP-793

Commit:
Create a Travis CI testing set-up for webinos

This commit adds a testing script (auto-test.sh) and then
references it from package.json.  It also adds a command-line
option to webinos_pzp and webinos_pzh such that they exit if
they have been started successfully (--test).  In addition, to
test the PZH, a travis-specific configuration file has been
added (travis_webinos_config.json) to change ports to high
numbers.  This needs fixing in the future.

Jira issue: WP-793

Commit:
Minor fix to travis.yml

Jira issue: WP-793

Commit:
Rebase of Travis CI integration work with new Grunt version.

Jira issue: WP-793

Commit:
Change Travis CI to use node v0.8 and fix dependency issues

Travis will now temporarily move the offending node_modules
files (e.g., expat.js) before installation.

Jira issue: WP-793

Commit:
Travis integration

This commit also adds another exit case in the PZP.

Jira issue: WP-793

Add webinos test script to Travis config

This commit adds an automatic test harness which looks for
Jasmine unit tests and executes them.  There's also a whitelist
to select just tests that currently work.

In addition, this version removes the temporary travis
version of webinos_config and adds another test script.

Jira issue: WP-793

Moved the travis scripts to the tools/travis directory.

To keep the root directory clean, the testing scripts are
now in ./tools/travis.  A little neatening of code and comments
was also included.

Jira issue: WP-793

habibvirji commented on webinos_config.json:
```diff
@@ -1,7 +1,7 @@
-    "provider"          :80,
-    "provider_webServer":443,
+    "provider"          :6080,
+    "provider_webServer":6443,
```
Any reason why ports are changed

jplyle replied:
That's been changed, now.

jplyle replied:
$%£! Some kind of rebase errors have brought this back in.

jplyle commented:
Wow, I've never submitted a pull request that needed rebasing quite so quickly...

Commit:
Travis CI integration: fixes for rebase errors.

Some of my previous changes got lost in a rebase.  This
patch reapplies them.  These are mostly formatting issues
rather than real problems, although I did manage to
lose some changes.

Jira issue: WP-793
